lexer.go

```diff
@@ -107,6 +107,11 @@ func longestMatch(source string, ic cursor, options []string) string {
 	cur := ic
 
 	for {
```

Comment:
Nice catch. Might make more sense to add into the for loop guard here?

```
for cur.pointer < uint(len(source)) {
   ...
```

Could you add a lexer test with your example to catch this?

Reply:
Done